File: 1 Python/solutions/pick6.py
from random import randint

def pick_six():
    return [randint(1, 99) for i in range(6)]
  
def compare_matches(winner, ticket):
    matches = 0
    for i in range(len(ticket)):
        if ticket[i] == winner[i]:
            matches += 1
    return matches
        
def main():
    payouts = [0, 4, 7, 5E4, 1E6, 25E7]
    MAX_ITER = int(1E5)
    balance = 0
    expenses = 0
    winner = pick_six()

    for i in range(MAX_ITER):
        expenses += 2
        matches = compare_matches(winner, pick_six())
        balance += payouts[matches]

    print(f'\n Balance: {balance}\n Expenses: {expenses} \n ROI: {(balance - expenses) / expenses}\n')

main()

File: 3 JavaScript/solutions/pick6.js
function generateTicket() {
    low = 1;
    high = 99;
    const ticket = [];
    for (let i = 0; i < 6; i++)
        ticket.push(Math.floor(Math.random() * (1 + high - low)) + low);
    return ticket;
}


function compareTickets(winner, ticket) {
    let matches = 0;
    winner.forEach(function (value, index) {
        if (value == ticket[index])
            matches++;
    })
    return matches;
}


function main() {
    const prizes = {
        0: 0,
        1: 4,
        2: 7,
        3: 100,
        4: 50000,
        5: 1000000,
        6: 25000000,
    }
    
    
    let matches = 0;
    let winnings = 0;
    let expenses = 0;
    let iter = 0;
    const MAX_ITER = 99999;
    let ticket = [];


    const winner = generateTicket();


    while (iter <= MAX_ITER) {
        expenses += 2;
        ticket = generateTicket();

        matches = compareTickets(winner, ticket);

        winnings += prizes[matches];
        iter++;
    }


    console.log(`\nwinnings: ${winnings}\nexpenses: ${expenses}\nroi: ${(winnings - expenses) / expenses}\n`)
}

main()